src/main/java/net/minestom/server/inventory/Inventory.java

```diff
@@ -46,41 +38,32 @@
     private final byte id;
     // the type of this inventory
     private final InventoryType inventoryType;
-    // the title of this inventory)
-    private String title;
-
-    // the size based on the inventory type
-    private final int size;
+    // the title of this inventory
+    private Component title;
 
     private final int offset;
 
-    // the items in this inventory
-    private final ItemStack[] itemStacks;
     // the players currently viewing this inventory
     private final Set<Player> viewers = new CopyOnWriteArraySet<>();
     private final Set<Player> unmodifiableViewers = Collections.unmodifiableSet(viewers);
     // (player -> cursor item) map, used by the click listeners
     private final ConcurrentHashMap<Player, ItemStack> cursorPlayersItem = new ConcurrentHashMap<>();
 
-    // list of conditions/callbacks assigned to this inventory
-    private final List<InventoryCondition> inventoryConditions = new CopyOnWriteArrayList<>();
-    // the click processor which process all the clicks in the inventory
-    private final InventoryClickProcessor clickProcessor = new InventoryClickProcessor();
-
-    private Data data;
-
-    public Inventory(@NotNull InventoryType inventoryType, @NotNull String title) {
+    public Inventory(@NotNull InventoryType inventoryType, @NotNull Component title) {
+        super(inventoryType.getSize());
         this.id = generateId();
         this.inventoryType = inventoryType;
         this.title = title;
 
-        this.size = inventoryType.getSize();
-
-        this.offset = size;
-
-        this.itemStacks = new ItemStack[size];
+        this.offset = getSize();
+    }
 
-        ArrayUtils.fill(itemStacks, ItemStack::getAirItem);
+    /**
+     * @deprecated use {@link Inventory#Inventory(InventoryType, Component)}
+     */
+    @Deprecated
+    public Inventory(@NotNull InventoryType inventoryType, @NotNull String title) {
+        this(inventoryType, Component.text(title));
```

Comment:
If this string was written directly, could it contain JSON? If so, this should be `GsonComponentSerializer.serializer().deserialize(title)`.